rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,261 @@
+
+To add the pipeline information to the task, we need to add a new optional parameter to the [`Task::new`](https://github.com/timberio/vector/blob/v0.15.0/src/topology/task.rs#L29) method.
+
+```rust
+pub struct Task {
+    #[pin]
+    inner: BoxFuture<'static, Result<TaskOutput, ()>>,
+    name: String,
+    typetag: String,
+    pipeline: Option<String>,
+}
+impl Task {
+    pub fn new<S1, S2, Fut>(name: S1, typetag: S2, pipeline: Option<String>, inner: Fut) -> Self
+    where
+        S1: Into<String>,
+        S2: Into<String>,
+        Fut: Future<Output = Result<TaskOutput, ()>> + Send + 'static,
+    {
+        Self {
+            inner: inner.boxed(),
+            name: name.into(),
+            typetag: typetag.into(),
+            pipeline,
+        }
+    }
+}
+````
+
+That way, when vector [spawns a new transform task](https://github.com/timberio/vector/blob/v0.15.0/src/topology/mod.rs#L574), it will be able to add the optional pipeline information to the span.
+
+```rust
+let span = error_span!(
+    "transform",
+    component_kind = "transform",
+    component_name = %task.name(),
+    component_type = %task.typetag(),
+    pipeline_id = %task.pipeline_id(),
+);
+```
+
+Doing so, each time the task will emit an internal event, it will be populated by the optional `pipeline_id`.
+
+## Rationale
+
+- Why is this change worth it?
+
+This split improves the readability of the configuration files and allows the users to collaborate, which makes using Vector more user-friendly.
+
+- What is the impact of not doing this?
+
+This would force the users to keep having complex configuration files and/or to duplicate components configuration between their configuration files.
+
+- How does this position us for success in the future?
+
+With this representation, we'll be able add access control by, for example, declaring the pipelines inside the configuration files to limit the reachable components. We would also be able to specify a quota for each pipeline.
+
+## Prior Art
+
+_TODO_
+
+## Drawbacks
+
+- Why should we not do this?
+- What kind of ongoing burden does this place on the team?
+
+## Alternatives
+
+- Do nothing: we can already use several configuration files, people could split their existing configuration.
+
+This would imply some duplication if a transform is used in multiple configuration files.
+Anybody could add a sink or source even if they don't have permission.
+
+- Do nothing: write a tool that would write a big configuration file where each pipeline would start with a dummy filter that we could monitor in the `internal_metrics`.
+
+
+
+- Evolve vector to use a tag/filter model like our competitors, have a 'pipeline' be a 'tag'.
+
+Not able to add internal metrics to specific transforms and monitor them.
```

Comment:
Can't tell whether this is a standalone bullet, or part of the previous?

Reply:
It's a standalone one, but the 3 first are kind of related and have similar answers.